02_bubble/nickolee/bubblesort.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// general sorting algorithm that sorts by "bubbling" smaller numbers to the left
+func bubbleSort(s []int) []int {
+	result := make([]int, len(s))
+	copy(result, s)
+
+	unsorted := true
```

Comment:
Usually you don't want negators in names for booleans. The reason is it makes it harder to read statements
eg:
`!unsorted` is the same as `sorted`
`for unsorted` is the same as `for !sorted`
`unsorted = false` is the same as `sorted = true`

In general, it just reads better and makes it easier to understand, even when theres more symbols (`!sorted` reads just as well as `unsorted`).

Reply:
you mean negators in the variable name itself right?

Reply:
so sorted = false

Reply:
Refactored to reflect this